### ISSUE TITLE
Smile Tracker throws exception when campaign variables are detected

### ISSUE THREAD

**Issue by TiEul** [Smile-SA/elasticsuite#584]
### Preconditions

Magento Version : 2.1.9 EE / Cloud

Environment : Production

### Steps to reproduce

Adding `?utm_source=something`, `?utm_campaign=something` or `?utm_medium=something` to the URL will cause the Smile tracking.js to throw the following error:

`Uncaught TypeError: Cannot read property 'addPageVar' of undefined`

The error is in the `addCampaignVars()` function and occurs in the following line:

`this.addPageVar(paramName, paramValue);` 

Apparently `this` does not work in this context. The issue occurs on every page and does not seem to be related to any special configuration of the environment. If you need a link to the environment, please let me know and I will send it to you in private.

Issue can be reproduced on the latest version of Chrome without any add-ons installed.

### Expected result
No errors should be thrown, campaign variables should be handled properly.

### Actual result
```
e0799d5c1ca507c99fc95216a78017af.js:3898 Uncaught TypeError: Cannot read property 'addPageVar' of undefined
    at e0799d5c1ca507c99fc95216a78017af.js:3898
    at Array.forEach (<anonymous>)
    at SmileTrackerImpl.addCampaignVars (e0799d5c1ca507c99fc95216a78017af.js:3893)
    at SmileTrackerImpl.getTrackerUrl (e0799d5c1ca507c99fc95216a78017af.js:3938)
    at SmileTrackerImpl.sendTag (e0799d5c1ca507c99fc95216a78017af.js:3965)
```

**Comment by romainruaud** [comment]
Confirmed. I can clearly see the same bug on our demo : http://demo.magento-elastic-suite.io/index.php/men/tops-men.html?utm_source=pipo

Dunno why it happens... I will try to provide a fix soon.

Regards.

**Comment by afoucret** [comment]
Fixed by PR #585 in branch master, 2.3.x and 2.4.x.
Will be part of the next releases.
